src/main/java/br/com/equipe7/desafio_spring/repository/ProductRepo.java

```diff
@@ -27,4 +28,26 @@ public ProductRepo() {
             System.out.println("Error reading file");
         }
     }
+
+    public Optional<Product> getProductById(int id) {
+        List<Product> products = loadProducts();
```

Comment:
Acho interessante pegar pelo constructor do Repository a lista de produtos, veja se faz sentido

```suggestion
        Optional<Product> product = productList.stream()
                .filter(p -> p.getProductId() == id)
                .findFirst();

        return product;
```